### ISSUE TITLE
Server: CurrentTick overflow (season bugs)

### ISSUE THREAD

**Issue by croxxx** [manicdigger/manicdigger#163]
After noticing that the seasons hadn't changed for a month (eternal spring) on my server Pixolia I decided to dig a little deeper.
As a quick and dirty workaround I changed season manually using my ServerUtilities Mod. It does exactly the same (or at least should) as the ingame automatic season changing. After this I had winter (as I wanted).
As I rejoined today, I noticed season had changed to spring again.
*In addition I had a real strange bug. Everything below water surface was dark. Water was handled solid for light? Changing seasons manually did not help. Maybe a bug in my code? Not sure here.*

After some investigation I decided to dump all time-related variables I could get my hands on via Mod-API. This is the result:
```
CurrentYearTotal: -338.973439127604
CurrentHourTotal: -8135.3625390625
CurrentTick: -1874387529
GameYearRealHours: 24
GameDayRealHours: 1
currentSeason: -3

CurrentYearTotal: -338.97342755353
CurrentHourTotal: -8135.36226128472
CurrentTick: -1874387465
GameYearRealHours: 24
GameDayRealHours: 1
currentSeason: -3

CurrentYearTotal: -338.973415979456
CurrentHourTotal: -8135.36198350694
CurrentTick: -1874387401
GameYearRealHours: 24
GameDayRealHours: 1
currentSeason: -3
```
It is obvious that this cannot work.
It seems like the CurrentTick variable overflowed. This results in faulty values for "current time". This value is currently stored as an integer (-2.147.483.648 to 2.147.483.647).

Shall this simply be fixed by checking for an overflow and resetting to 0?

**Comment by TobiMob** [comment]
Those games tick are mostly used for seasons ?
Maybe changing the type to uint32 could help for now, it would automatically overflow to 0.

On the long run I would like to see something like a [Timespan](https://msdn.microsoft.com/en-us/library/system.timespan(v=vs.110).aspx) (which can be serialized as a long) to handle the Time independent from the server ticks.

After Time and ServerTicks are decoupled, maybe there would be no more need to store/load the ServerTicks?

**Comment by croxxx** [comment]
Should be fixed by 468d38e9d5c69d4d1676db900cf8f2c9f568807b and following commits.
